spec/unit/provider/service/systemd_spec.rb

```diff
@@ -372,6 +372,29 @@
     end
   end
 
+  describe "#insync_enabled?" do
+    provider = described_class.new(Puppet::Type.type(:service).new(:name => 'sshd.service', :enable => false))
+
+    it "should consider the enable state of a static service to be always in sync and print a debug log" do
+      expect(provider).to receive(:cached_enabled?).and_return('static')
+      expect(Puppet).to receive(:debug).with("Unable to enable or disable static service sshd.service")
+      expect(provider.enabled_insync?(true)).to eq(true)
+    end
+    
+    it "should consider the enable state of an indirect service to be always in sync and print a warning log" do
+      expect(provider).to receive(:cached_enabled?).and_return('indirect')
+      expect(Puppet).to receive(:warning).with("Service sshd.service is in 'indirect' state and cannot be enabled/disabled")
+      expect(provider.enabled_insync?(true)).to eq(true)
+    end
+
+    it "should check normally and return if in sync (when the state is not static nor indirect) and print no debug or warning logs" do
+      expect(provider).to receive(:cached_enabled?).and_return('enabled')
+      expect(Puppet).not_to receive(:debug)
+      expect(Puppet).not_to receive(:warning)
+      expect(provider.enabled_insync?(true)).to eq(false)
+    end
```

Comment:
[Optional] I would split this tests using context to emphasise on each case case that is tested, something like:
```ruby

  describe "#insync_enabled?" do
    let(:provider) do
      described_class.new(Puppet::Type.type(:service).new(:name => 'sshd.service', :enable => false))
    end

    before do
      allow(provider).to receive(:cached_enabled?).and_return(service_state)
    end

    context 'when service is static' do
      let(:service_state) { 'static' }

      it 'is enabled_insync' do
        expect(provider).to be_enabled_insync(true)
      end

      it 'logs a debug messsage' do
        expect(Puppet).to receive(:debug).with("Unable to enable or disable static service sshd.service")
        provider.enabled_insync?(true)
      end
    end
  end
```

I know this is not the same on how other tests in this file are written but they are easier to read and understand this way